### DISCUSSION TITLE
querying multiple joined tables

### DISCUSSION THREAD

**Discussion by timdiacon** [supabase/supabase#11318] [Questions]
I have the following (abbreviated) tables

Profiles

| id | 
|---|
| 1 |
| 2 |
| 3 | 

Subscriptions
| id | trader_id | profile_id|
|---|---|---|
| 1 | 1 | 1 |
| 2 | 1| 2|

Orders
|id| trader_id | profile_id |
|---|---|---|
| 1 | 1 | 2 |
| 2 | 1 | 3 |

I want to select all the profiles which have either a subscription or order (or both) which contain a specified value for the trader_id. The solution I have come up with is to use two separate queries and then combine them on the client but the OCD side of me really wants to find a way of doing it in a single query.

`client.from('profiles').select(id, subscriptions!inner(trader_id)).eq('subscriptions.trader_id', 1)`

`client.from('profiles').select(id, orders!inner(trader_id)).eq('orders.trader_id', 1)`

The problem I'm running into is if I join both tables in the select:

`select(id, subscriptions!inner(trader_id), orders!inner(trader_id))`

I can only retrieve profiles that have **both** an order **and** a subscription as the `!inner` statement filters out any profiles without either a subscription or profile.

Is it possible to do this in a single query?

**Comment by steve-chavez** [comment]
> Is it possible to do this in a single query?

Yes, for now this can be done locally with the Supabase CLI. Like:

```js
client
.from('profiles')
.select('id, subscriptions(trader_id), orders(trader_id)')
.eq('subscriptions.trader_id', 1)
.eq('orders.trader_id', 1)
.or('subscriptions.not.is.null,orders.not.is.null')
```

---

Doing the `not.is.null` filters on the related tables is more flexible than `!inner`. There was a mention of this on https://supabase.com/blog/postgrest-11-prerelease#anti-joins (last paragraph of the anti joins section).

**Reply by timdiacon** [reply]
Thanks so much for your reply, this looks like exactly what I'm after. When I try it however I get an error

`column profiles.subscriptions does not exist`

If I use a .not instead of the .or such as `.not('orders', 'is', null)` again I get an error of `column profiles.orders does not exist`

:-(

**Reply by GaryAustin1** [reply]
Are you using a local docker instance?  I believe this requires PostgREST11 (per Steve's link) whichI don't think is on hosted instances yet.
